### PR TITLE
adds haircutting skillchip and trait, changes haircutting time.

### PR DESCRIPTION
**Pull request by capsaicinz**
## About The Pull Request

adds a haircutting trait that reduces haircutting time by a decent amount, along with a skillchip that uses this trait. also reduces the default time of hairstyle changes to 45 seconds. there's an extra one in the barber's locker.

**WITH SKILLCHIP:**

45 -> **30** (hairstyle)
20 -> **15** (facial hair)

## How This Contributes To The Skyrat Roleplay Experience

haircutting took a minute for **EVERYBODY** before. barbers should be a little better at it if their entire job is focused around it- a minute is a lot to stand still uninterrupted.

## Changelog
:cl:
add: haircutting trait and skillchip!
balance: default haircutting time has been reduced.
/:cl:

### CODE REVIEW

File: code/__DEFINES/traits.dm [hunk 1]
```diff
@@ -427,6 +427,9 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_ENTRAILS_READER "entrails_reader"
 /// this skillchip trait lets you wash brains in washing machines to heal them
 #define TRAIT_BRAINWASHING "brainwashing"
+// SKYRAT EDIT START - Barber Skillchip
+#define TRAIT_BARBER "barber"
+// SKYRAT EDIT END
```
Comment (jjpark-kb) [inline code/__DEFINES/traits.dm]: We have a skyrat defines folder somewhere.

File: code/_globalvars/traits.dm [hunk 2]
```diff
@@ -135,6 +135,7 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_LIGHTBULB_REMOVER" = TRAIT_LIGHTBULB_REMOVER,
 		"TRAIT_KNOW_CYBORG_WIRES" = TRAIT_KNOW_CYBORG_WIRES,
 		"TRAIT_KNOW_ENGI_WIRES" = TRAIT_KNOW_ENGI_WIRES,
+		"TRAIT_BARBER" = TRAIT_BARBER, //SKYRAT EDIT ADDITION - Barber Skillchip
```
Comment (jjpark-kb) [inline code/_globalvars/traits.dm:138]: same thing as above comment

File: modular_skyrat/modules/salon/code/scissors.dm [hunk 3]
```diff
@@ -70,8 +72,14 @@
 
 		playsound(target_human, 'modular_skyrat/modules/salon/sound/haircut.ogg', 100)
 
-		if(do_after(user, facial_haircut_duration, target_human))
+		if(do_after(user, (HAS_TRAIT(user, TRAIT_BARBER) ? FACIAL_HAIRCUT_SPEED_BARBER : FACIAL_HAIRCUT_SPEED_NONBARBER), target_human))
 			target_human.facial_hairstyle = facial_hair_id
 			target_human.update_hair()
-			user.visible_message(span_notice("[user] successfully cuts [target_human]'s facial hair!"), span_notice("You successfully cut [target_human]'s facial hair!"))
+			user.visible_message(span_notice("[user] gracefully cuts [target_human]'s facial hair!"), span_notice("You gracefully cut [target_human]'s facial hair!"))
```
Comment (jjpark-kb) [inline modular_skyrat/modules/salon/code/scissors.dm]: you still keeping this?

File: modular_skyrat/modules/salon/code/scissors.dm [hunk 4]
```diff
@@ -49,10 +50,11 @@
 
 		playsound(target_human, 'modular_skyrat/modules/salon/sound/haircut.ogg', 100)
 
-		if(do_after(user, haircut_duration, target_human))
+		if(do_after(user, (HAS_TRAIT(user, TRAIT_BARBER) ? HAIRCUT_SPEED_BARBER : HAIRCUT_SPEED_NONBARBER), target_human))
```
Comment (Zonespace27) [inline modular_skyrat/modules/salon/code/scissors.dm]: early return

File: modular_skyrat/modules/salon/code/scissors.dm [hunk 5]
```diff
@@ -70,8 +72,14 @@
 
 		playsound(target_human, 'modular_skyrat/modules/salon/sound/haircut.ogg', 100)
 
-		if(do_after(user, facial_haircut_duration, target_human))
+		if(do_after(user, (HAS_TRAIT(user, TRAIT_BARBER) ? FACIAL_HAIRCUT_SPEED_BARBER : FACIAL_HAIRCUT_SPEED_NONBARBER), target_human))
```
Comment (Zonespace27) [inline modular_skyrat/modules/salon/code/scissors.dm]: early return

File: modular_skyrat/modules/salon/code/scissors.dm [hunk 6]
```diff
@@ -70,8 +72,14 @@
...
 			target_human.facial_hairstyle = facial_hair_id
 			target_human.update_hair()
-			user.visible_message(span_notice("[user] successfully cuts [target_human]'s facial hair!"), span_notice("You successfully cut [target_human]'s facial hair!"))
+			user.visible_message(span_notice("[user] gracefully cuts [target_human]'s facial hair!"), span_notice("You gracefully cut [target_human]'s facial hair!"))
+			user.balloon_alert_to_viewers("snip snippity!")
```
Comment (Zonespace27) [inline modular_skyrat/modules/salon/code/scissors.dm]: against the intended purpose of balloon alerts, they're supposed to be quick and concise ways of telling what's happened/happening

File: modular_skyrat/modules/salon/code/scissors.dm [hunk 7]
```diff
@@ -49,10 +50,11 @@
 
 		playsound(target_human, 'modular_skyrat/modules/salon/sound/haircut.ogg', 100)
 
-		if(do_after(user, haircut_duration, target_human))
+		if(do_after(user, (HAS_TRAIT(user, TRAIT_BARBER) ? HAIRCUT_SPEED_BARBER : HAIRCUT_SPEED_NONBARBER), target_human))
 			target_human.hairstyle = hair_id
 			target_human.update_hair()
 			user.visible_message(span_notice("[user] successfully cuts [target_human]'s hair!"), span_notice("You successfully cut [target_human]'s hair!"))
+			user.balloon_alert_to_viewers("snippity snip!")
```
Comment (Zonespace27) [inline modular_skyrat/modules/salon/code/scissors.dm]: ditto

File: modular_skyrat/modules/salon/code/misc_items.dm [hunk 8]
```diff
@@ -168,3 +168,13 @@
...
+
+/obj/item/skillchip/haircut
+	name = "FR3SH-CUT skillchip"
+	desc = "I like your cut, G!"
+	auto_traits = list(TRAIT_BARBER)
+	skill_name = "Haircutting Proficiency"
+	skill_description = "Learn to get yourself out of hair-y situations, by mastering the art of cutting through it!"
+	skill_icon = "hand-scissors"
+	activate_message = span_notice("In your mastery of haircuts, you feel ready to shave the day!")
+	deactivate_message = span_notice("You feel your mastery of haircuts get cut short..")
```
Comment (axietheaxolotl) [inline modular_skyrat/modules/salon/code/misc_items.dm:180]:
```suggestion
	deactivate_message = span_notice("You feel your mastery of haircuts get cut short...")
```

Comment (capsaicinz) [reply modular_skyrat/modules/salon/code/misc_items.dm:180]: never

**Comment by capsaicinz** [comment]
being made obsolete by goof's barber pr, if still needed we can look at doin this again soon

**Pull request closed without merging**